libs/dal/src/repositories/integration/integration.repository.ts

```diff
@@ -43,13 +44,36 @@ export class IntegrationRepository extends BaseRepository<IntegrationDBModel, In
 
     return await super.create(data);
   }
+
   async delete(query: IntegrationQuery) {
     const integration = await this.findOne({ _id: query._id, _environmentId: query._environmentId });
     if (!integration) throw new DalException(`Could not find integration with id ${query._id}`);
 
     await this.integration.delete({ _id: integration._id, _environmentId: integration._environmentId });
   }
 
+  async deleteMany(query: IntegrationQuery): Promise<IDeleteResult> {
+    const { _environmentId, _organizationId } = query || {};
+    if (!_environmentId || !_organizationId) {
+      throw new DalException(
+        'Deletion operation blocked for missing any of these properties: [_environmentId, _organizationId]. We are avoiding a potential unexpected multiple deletion'
+      );
+    }
+
+    const { acknowledged, modifiedCount, matchedCount } = await this.integration.delete(query);
+
+    if (matchedCount === 0 || modifiedCount === 0) {
+      throw new DalException(
```

Comment:
Would not it be enough to return the actual result of matchedCount = 0 modifiedCount = 0?

Reply:
I would expect that if trying to delete entries and there are not matched or any of the matched updated that means the query is like a 404 for the first case and for the second an error has happened below the DAL level, therefore is important to flag that it happened.
Do you think returning kinda like a go through in this method serves better purposes? Also that for me would mean the DAL responsibility of the management of what is happening is pushed in other layers. 😕

Reply:
tend to agree maybe we could add it in the DAL layer (most definitely not in this PR)